openupgrade_scripts/scripts/account/15.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,429 @@
+            sequence INTEGER,
+            CONSTRAINT account_payment_method_line_pkey PRIMARY KEY (id)
+        )""",
+    )
+    # Create account payment method lines from account payment methods
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_payment_method_line
+        (name, payment_method_id, journal_id)
+        SELECT apm.name, apm.id, aj.id
+        FROM account_payment_method apm
+        JOIN account_journal aj ON aj.type IN ('bank', 'cash')
+        WHERE apm.code = 'manual'
+        """,
+    )
+
+
+def _fast_fill_account_payment_payment_method_line_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_payment
+        ADD COLUMN IF NOT EXISTS payment_method_line_id INTEGER""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_payment ap
+        SET payment_method_line_id = apml.id
+        FROM account_move am
+        JOIN account_payment_method_line apml ON apml.journal_id = am.journal_id
+        WHERE ap.move_id = am.id
+        """,
+    )
+
+
+def _fill_account_tax_country_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_tax
+        ADD COLUMN IF NOT EXISTS country_id INTEGER
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_tax at
+        SET country_id = p.country_id
+        FROM res_company c
+        JOIN res_partner p ON p.id = c.partner_id
+        WHERE c.id = at.company_id
+        """,
+    )
+
+
+def _fill_res_company_account_journal_payment_credit_account_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE res_company
+        ADD COLUMN IF NOT EXISTS account_journal_payment_credit_account_id INTEGER
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE res_company c
+        SET account_journal_payment_credit_account_id = (
+        SELECT aj.payment_credit_account_id
+        FROM account_journal aj
+        WHERE aj.company_id = c.id AND aj.type IN ('bank', 'cash')
+            AND aj.payment_credit_account_id IS NOT NULL
+        LIMIT 1)
+        """,
+    )
+
+
+def _fill_res_company_account_journal_payment_debit_account_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE res_company
+        ADD COLUMN IF NOT EXISTS account_journal_payment_debit_account_id INTEGER
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE res_company c
+        SET account_journal_payment_debit_account_id = (
+        SELECT aj.payment_debit_account_id
+        FROM account_journal aj
+        WHERE aj.company_id = c.id AND aj.type IN ('bank', 'cash')
+            AND aj.payment_debit_account_id IS NOT NULL
+        LIMIT 1)
+        """,
+    )
+
+
+def _fast_fill_account_payment_outstanding_account_id(env):
```

Comment:
But I think we should only create an empty `outstanding_account_id` column at this pre-migration, and then fill the values at end-migration, after all the payment method lines are created (from other module like `account_check_printing` and `payment_xxx`) and assigned to the `payment_method_line_id` field on payments.